rust/cubesql/cubesql/e2e/tests/postgres.rs

```diff
@@ -584,6 +584,126 @@ impl PostgresIntegrationTestSuite {
             }
         ).await?;
 
+        self.test_simple_query(
+            r#"declare test_cursor_fetching_less_than_batch_size cursor with hold for SELECT * from information_schema.testing_dataset order by id;"#
+                .to_string(),
+            |messages| {
+                assert_eq!(messages.len(), 1);
+            }
+        ).await?;
+
+        self.test_simple_query(
+            r#"fetch 800 in test_cursor_fetching_less_than_batch_size; fetch 800 in test_cursor_fetching_less_than_batch_size; fetch 5000 in test_cursor_fetching_less_than_batch_size;"#
+                .to_string(),
+            |messages| {
+                // 5000 rows | 3 completions
+                assert_eq!(messages.len(), 5003);
+
+                for i in 0..800 {
+                    if let SimpleQueryMessage::Row(row) = &messages[i] {
+                        assert_eq!(row.get(0), Some(i.to_string().as_str()));
+                    } else {
+                        panic!("Must be Row command, {}", i)
+                    }
+                }
+
+                if let SimpleQueryMessage::CommandComplete(rows) = messages[800] {
+                    assert_eq!(rows, 800_u64);
```

Comment:
It's too crazy to check every row. Let's assert the first and last row. + let's introduce a function to check that instead of copy/pasting